src/Compilers/CSharp/Portable/Binder/Binder_InterpolatedString.cs

```diff
@@ -204,6 +204,10 @@ bool tryBindAsHandlerType([NotNullWhen(true)] out BoundInterpolatedString? resul
             }
         }
 
+        private bool ValidateInterpolatedStringParts(BoundUnconvertedInterpolatedString unconvertedInterpolatedString)
+            => !unconvertedInterpolatedString.Parts.ContainsAwaitExpression()
+               && unconvertedInterpolatedString.Parts.All(p => p is not BoundStringInsert { Value: { Type: { TypeKind: TypeKind.Dynamic } } });
```

Comment:
nit: consider extended property pattern to compact a bit `Value.Type.TypeKind: ...`

Reply:
Sorry, ignore that. It will run into code style issue (my PR wasn't merged yet)